### ISSUE TITLE
[NEWS] Wysyłam z Allegro - 31 stycznia 2025 usuniemy pole "streetNumber" / Ship with Allegro -  on January 31, 2025 we will remove the "streetNumber" field

### ISSUE THREAD

**Issue by AureliuszBrussy** [allegro/allegro-api#10061]
W ramach [Wysyłam z Allegro](https://help.allegro.com/sell/pl/a/czym-jest-narzedzie-wysylam-z-allegro-i-do-czego-sluzy-oAd1MRDd2t6), oznaczyliśmy dzisiaj jako **deprecated** pole ``"streetNumber"``, znajdujące się w obiektach:

- ``"sender"``
- ``"receiver"``
- ``"pickup"``.
 
**31 stycznia 2025** całkowicie je usuniemy. Dotyczy to endpointów: 

- [POST /shipment-management/shipments/create-commands](https://developer.allegro.pl/documentation#operation/createNewShipment)
- [GET /shipment-management/shipments/{shipmentId}](https://developer.allegro.pl/documentation#operation/getShipmentDetails).

**Co to oznacza i jak przygotować się do zmiany?**

Od dzisiaj, cały adres, czyli nazwę ulicy oraz jej numer, możesz przekazać w polu ``"street"``. Wciąż możesz korzystać z pola ``"streetNumber"`` jak dotychczas, jednakże, **31 stycznia 2025 roku**, całkowicie je usuniemy . Zadbaj do tego czasu, aby nazwę ulicy wraz z numerem we wszystkich obiektach wysyłać w polu ``"street"``.

**Dlaczego wprowadziliśmy zmianę?**

Poprzez uproszczenie sposobu przekazania adresu ułatwiamy proces tworzenia przesyłek. Generując przesyłkę unikniesz konieczności parsowania adresu, i tym samym przekażesz wartość w niezmiennej formie, po pobraniu jej z pola ``"delivery.address.street"`` w odpowiedzi dla:  

- [GET /order/checkout-forms/{id}](https://developer.allegro.pl/documentation#operation/getOrdersDetailsUsingGET)
- [GET /order/checkout-forms](https://developer.allegro.pl/documentation#operation/getListOfOrdersUsingGET).

Więcej o Wysyłam z Allegro przeczytasz [w naszym poradniku](https://developer.allegro.pl/tutorials/jak-zarzadzac-przesylkami-przez-wysylam-z-allegro-LRVjK7K21sY).

-------

Within [Ship with Allegro](https://help.allegro.com/sell/pl/a/czym-jest-narzedzie-wysylam-z-allegro-i-do-czego-sluzy-oAd1MRDd2t6), today we have marked the ``"streetNumber"`` field as **deprecated**. This field is located in objects: 

- ``"sender"``
- ``"receiver"``
- ``"pickup"``.

On **January 31, 2025**, we will remove this field completely. This applies to endpoints: 

- [ POST /shipment-management/shipments/create-commands](https://developer.allegro.pl/documentation#operation/createNewShipment)
- [GET /shipment-management/shipments/{shipmentId}](https://developer.allegro.pl/documentation#operation/getShipmentDetails).

**What does this mean and how to prepare for the change?**

From today, you can pass the entire address, that is, the street name and its number, in the ``"street"`` field. You can still use the ``"streetNumber"`` field as before, however, on **January 31, 2025**, we will remove it completely. Until then, prepare to send the street name along with the number in all objects in the ``"street"`` field.

**Why are we making the change?**

By simplifying the way the address is passed, we make the process of creating shipments easier. When generating a shipment, you will avoid the need to parse the address, and thus pass the value in an immutable form, after retrieving it from the ``"delivery.address.street"`` field in the response for:

- [GET /order/checkout-forms/{id}](https://developer.allegro.pl/documentation#operation/getOrdersDetailsUsingGET)
- [GET /order/checkout-forms](https://developer.allegro.pl/documentation#operation/getListOfOrdersUsingGET).  

You can read more about Ship from Allegro [in our guide](https://developer.allegro.pl/tutorials/how-to-manage-parcels-via-ship-with-allegro-ZM9YAyGKWTV).

**Comment by Wojtu204** [comment]
Proszę o informację, czy w związku z tym zostaną wprowadzone na formularzu zamówień ograniczenie/walidacja w długości podawanego przez kupującego adresu?
Już teraz mamy problemy z nadawaniem przesyłek - API kuriera odrzuca żądanie, bo pole z ulicą w zgłoszeniu paczki jest za długie - przekracza 35 znaków, z kolei w danych zamówienia z Allegro w ulicy dostajemy nawet po 90 znaków, przykład w zamówieniu 589f6cc2-86f4-11ef-87d9-2df4fbef5083
Taki brak ograniczenia i walidacji zgodnej z "najostrzejszymi" wymogami kuriera powoduje problemy z obsługą zamówień i nadawaniem przesyłek - każdorazowo paczka musi być wstrzymana do czasu wyjaśnienia i poprawy danych adresowych.

**Comment by AureliuszBrussy** [comment]
@Wojtu204 nie, na ten moment nie planujemy zmieniać długości wprowadzonych przez kupującego danych, więc aktualnie pozostaje w takich przypadkach interwencja człowieka.
